sdcm/sct_runner.py

```diff
@@ -62,185 +87,285 @@ def instance_root_disk_size(test_duration) -> int:
-            sys.exit(1)
-        lt_datetime = datetime.datetime.now(tz=pytz.utc)
+            return None
         return self._create_instance(
             instance_type=instance_type or self.instance_type(test_duration=test_duration),
             base_image=self._get_base_image(self.image),
-            tags_list=[
-                {"Key": "Name", "Value": self.RUNNER_NAME},
-                {"Key": "TestId", "Value": test_id},
-                {"Key": "NodeType", "Value": self.NODE_TYPE},
-                {"Key": "RunByUser", "Value": get_username()},
-                {"Key": "keep", "Value": str(ceil(test_duration / 60) + 6)},
-                {"Key": "keep_action", "Value": "terminate"},
-                {"Key": "launch_time", "Value": lt_datetime.strftime("%B %d, %Y, %H:%M:%S")},
-            ],
+            tags={
+                "TestId": test_id,
+                "NodeType": self.NODE_TYPE,
+                "RunByUser": get_username(),
+                "keep": str(ceil(test_duration / 60) + 6),  # keep SCT Runner for 6h more than test_duration
+                "keep_action": "terminate",
+            },
             instance_name=f"{self.image_name}-instance-{test_id[:8]}",
-            region_az=region_az,
+            region_az=self.region_az(
+                region_name=self.region_name,
+                availability_zone=self.availability_zone,
+            ),
             test_duration=test_duration,
         )
 
 
 class AwsSctRunner(SctRunner):
-    """Provisions and configures the SCT runner on AWS"""
+    """Provision and configure the SCT Runner on AWS."""
     BASE_IMAGE = "ami-0ffac660dd0cb2973"  # ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20200609
     SOURCE_IMAGE_REGION = "eu-west-2"  # where the source Runner image will be created and copied to other regions
+    IMAGE_BUILDER_INSTANCE_TYPE = "t2.small"
+    REGULAR_TEST_INSTANCE_TYPE = "t3.large"  # has 7h 12m CPU burst
+    LONGTERM_TEST_INSTANCE_TYPE = "r5.large"
 
-    def __init__(self, region_name: str, availability_zone: str, cloud_provider: str = 'aws'):
-        super().__init__(cloud_provider=cloud_provider, region_name=region_name)
-        self.availability_zone = availability_zone
-        self.ec2_client = boto3.client("ec2", region_name=region_name)
-        self.ec2_client_source = boto3.client("ec2", region_name=self.SOURCE_IMAGE_REGION)
-        self.ec2_resource = boto3.resource("ec2", region_name=region_name)
-        self.ec2_resource_source = boto3.resource("ec2", region_name=self.SOURCE_IMAGE_REGION)
+    def __init__(self, region_name: str, availability_zone: str, cloud_provider: str = "aws"):
+        super().__init__(cloud_provider=cloud_provider, region_name=region_name, availability_zone=availability_zone)
+        self.aws_region = AwsRegion(region_name=region_name)
+        self.aws_region_source = AwsRegion(region_name=self.SOURCE_IMAGE_REGION)
```

Comment:
Do we really need "aws_" prefix here? To distinguish this implementation of the SCT runner from other we may use "cloud_provider" attribute.

Reply:
I prefer to stay with these names for a while because it's not uniformed how we work with different clouds. I've implemented it for Azure and switched for AWS, but GCE code still need to be refactored in this way. And, as I know, there is some work which should be done by @dkropachev in this direction too. Don't want to pay to much time on this for now.

Reply:
agree let's keep the naming consistent (keep the prefix). Dmitriy will rework this anyway